src/AppBundle/Resources/public/js/app.deck.js

```diff
@@ -24,7 +24,77 @@
             header_tpl = _.template('<h5><span class="icon icon-<%= code %>"></span> <%= name %> (<%= quantity %>)</h5>'),
             card_line_tpl = _.template('<span class="icon icon-<%= card.type_code %> fg-<%= card.faction_code %>"></span> <a href="<%= card.url %>" class="card card-tip" data-toggle="modal" data-remote="false" data-target="#cardModal" data-code="<%= card.code %>"><%= card.label %></a>'),
             layouts = {},
-            layout_data = {};
+            layout_data = {},
+            // restricted list v2.0, 2018-10-08
+            joust_restricted_list = [
+                "01100",
+                "01109",
+                "03038",
+                "04001",
+                "04017",
+                "05010",
+                "05049",
+                "06004",
+                "06039",
+                "06040",
+                "06098",
+                "06100",
+                "06103",
+                "09001",
+                "09017",
+                "09023",
+                "10045",
+                "10050"
+            ],
+            melee_restricted_list = [
+                "01001",
+                "01013",
+                "01043",
+                "01078",
+                "01119",
+                "01162",
+                "02012",
+                "02024",
+                "02060",
+                "03003",
+                "04003",
+                "04118",
+                "05001",
+                "05010",
+                "06004",
+                "06039",
+                "06098",
+                "07036",
+                "08098",
+                "09028"
+            ];
+
+
+    /*
+     * Validates the current deck against a given list of restricted cards.
+     * @param {Array} restricted_list
+     * @return {boolean}
+     */
+    var validate_deck_against_restricted_list = function(restricted_list) {
+        var is_valid = true;
+        var cards = app.deck.get_cards();
```

Comment:
Same question about agendas as the "slots" question server side.

Reply:
this holds up client-side as well. see comments and example above.